### ISSUE TITLE
kcov --collect-only and kcov --report-only

### ISSUE THREAD

**Issue by yoguo** [SimonKagstrom/kcov#232]
I used 'kcov --collect-only  /tmp/kcov ...' and 'kcov --report-only  /tmp/kcov ...' instead of  'kcov  /tmp/kcov ...'. But I found the result was not  the same in two methods. When using --collect-only and --report-only options, some files are not covered. And the coverage of the same file is not the same. I think the results of these two methods should be the same exactly.

**Comment by SimonKagstrom** [comment]
Yes, they should be the same. What are the differences you're seeing?

I can reproduce it myself when running kcov on itself. Will look into it.

**Comment by yoguo** [comment]
![kcov-collect-report](https://user-images.githubusercontent.com/31300413/35276159-626790c0-007d-11e8-96c1-bf00bd5385b0.png)
![kcov](https://user-images.githubusercontent.com/31300413/35276167-6c74f4e0-007d-11e8-9da4-68a6f190d64a.png)
The first picture shows the result when I run kcov --collect-only and kcov --report-only commands.
And the second one shows the result after running kcov /tmp/kcov command.

**Comment by yoguo** [comment]
Thank you for quick reply. The entire command: kcov  /tmp/kcov  virt-cat -a /home/RHEL-Server-7.4-64-20170711.0-hvm.raw /etc/fstab
If you want to use virt-cat, you have to install libguestfs-* packages on RHEL or Centos or fedora.

**Comment by SimonKagstrom** [comment]
Hmm... I'm looking at coverage of `kcov` itself, which really only differs in global constructors - the rest is the same for both "normal" and collect-only mode. That is strange as such, but your program looks much worse.

Are the code contained in shared libraries? If you run with `--skip-solibs` in addition to the other options, how does it look then?

I'll see if I can get a better example to reproduce this with.

**Comment by yoguo** [comment]
Sorry for late reply. The code are not contained in shared libraries. I tried --skip-solibs option, but unfortunately it had no effect.

**Comment by SimonKagstrom** [comment]
I took a look at this again, and at least in my case, the output database (coverage.db) is the same for both normal mode and `--collect-only`. `--report-only` writes it, which I think is actually a bug, but I don't think it should cause these symptoms since that happens after the report has been created.
